vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -511,6 +511,7 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
             updatedSelectedItem = true;
           }
         }
+        detailsOpenedItems = detailsOpenedItems.filter(openedItem => grid._getItemIndexInArray(openedItem, items) >= 0);
```

Comment:
This was the easiest change, I can also give it a try to rewrite the existing loop above and include this filter somehow.